SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizer.cc

```diff
@@ -0,0 +1,334 @@
+    crossingSimHitIndexOffset_.clear();
+  
+    iSetup.get<TrackerDigiGeometryRecord>().get(geometryType_, pDD_);
+    iSetup.get<IdealMagneticFieldRecord>().get(pSetup_);
+    iSetup.get<IdealGeometryRecord>().get(tTopoHand);
+    
+    // FIX THIS! We only need to clear and (re)fill this map when the geometry type IOV changes.  Use ESWatcher to determine this.
+    if (true) { // Replace with ESWatcher 
+      detectorUnits_.clear();
+      for (auto const & det_u : pDD_->detUnits()) {
+	unsigned int detId_raw = det_u->geographicalId().rawId();
+	DetId detId = DetId(detId_raw);
+	if (DetId(detId).det() == DetId::Detector::Tracker) {
+	  const Phase2TrackerGeomDetUnit* pixdet = dynamic_cast<const Phase2TrackerGeomDetUnit*>(det_u);
+	  assert(pixdet);
+	  detectorUnits_.insert(std::make_pair(detId_raw, pixdet));
+	}
+      }
+    }
+  }  
+  void 
+  Phase2TrackerDigitizer::accumulate(edm::Event const& iEvent, edm::EventSetup const& iSetup) {
+    accumulate_local<edm::Event>(iEvent, iSetup);
+  }
+
+  void  
+  Phase2TrackerDigitizer::accumulate(PileUpEventPrincipal const& iEvent, edm::EventSetup const& iSetup, edm::StreamID const& ) {
+    accumulate_local<PileUpEventPrincipal>(iEvent, iSetup);
+  }
+
+  template <class T>
+  void Phase2TrackerDigitizer::accumulate_local(T const& iEvent, edm::EventSetup const& iSetup) {
+    for (auto const & v : trackerContainers_) {
+      edm::Handle<std::vector<PSimHit> > simHits;
+      edm::InputTag tag(hitsProducer_, v);
+      iEvent.getByLabel(tag, simHits);
+
+      //edm::EDGetTokenT< std::vector<PSimHit> > simHitToken_(consumes< std::vector<PSimHit>(tag));
+      //iEvent.getByToken(simHitToken_, simHits);
+
+      unsigned int tofBin = PixelDigiSimLink::LowTof;
+      if (v.find(std::string("HighTof")) != std::string::npos) tofBin = PixelDigiSimLink::HighTof;
+      accumulatePixelHits(simHits, crossingSimHitIndexOffset_[tag.encode()], tofBin);
+      // Now that the hits have been processed, I'll add the amount of hits in this crossing on to
+      // the global counter. Next time accumulateStripHits() is called it will count the sim hits
+      // as though they were on the end of this collection.
+      // Note that this is only used for creating digi-sim links (if configured to do so).
+      if (simHits.isValid()) crossingSimHitIndexOffset_[tag.encode()] += simHits->size();
+     }
+  }
+  void Phase2TrackerDigitizer::finalizeEvent(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+    const bool isOuterTrackerReadoutAnalog = iconfig_.getParameter<bool>("isOTreadoutAnalog");
+    //Decide if we want analog readout for Outer Tracker.
+    addPixelCollection(iEvent, iSetup, isOuterTrackerReadoutAnalog);
+    if(!isOuterTrackerReadoutAnalog)
+      addOuterTrackerCollection(iEvent, iSetup);
+  }
+  void Phase2TrackerDigitizer::beginRun(edm::Run const& run, edm::EventSetup const& iSetup) {
+  }
+  std::string Phase2TrackerDigitizer::getAlgoType(unsigned int detId_raw) {
```

Comment:
it will even simplify the code in this case…

> On Feb 17, 2016, at 10:36 AM, Vincenzo Innocente <notifications@github.com> wrote:
> 
> In SimTracker/SiPhase2Digitizer/plugins/Phase2TrackerDigitizer.cc:
> 
> > +      // the global counter. Next time accumulateStripHits() is called it will count the sim hits
> > +      // as though they were on the end of this collection.
> > +      // Note that this is only used for creating digi-sim links (if configured to do so).
> > +      if (simHits.isValid()) crossingSimHitIndexOffset_[tag.encode()] += simHits->size();
> > +     }
> > +  }
> > +  void Phase2TrackerDigitizer::finalizeEvent(edm::Event& iEvent, const edm::EventSetup& iSetup) {
> > +    const bool isOuterTrackerReadoutAnalog = iconfig_.getParameter<bool>("isOTreadoutAnalog");
> > +    //Decide if we want analog readout for Outer Tracker.
> > +    addPixelCollection(iEvent, iSetup, isOuterTrackerReadoutAnalog);
> > +    if(!isOuterTrackerReadoutAnalog)
> > +      addOuterTrackerCollection(iEvent, iSetup);
> > +  }
> > +  void Phase2TrackerDigitizer::beginRun(edm::Run const& run, edm::EventSetup const& iSetup) {
> > +  }
> > +  std::string Phase2TrackerDigitizer::getAlgoType(unsigned int detId_raw) {
> 
> no, get rid of the string, use enum
> 
> —
> Reply to this email directly or view it on GitHub.
>